lib/whitehall_importer/integrity_checker/unpublishing_check.rb

```diff
@@ -0,0 +1,56 @@
+module WhitehallImporter
+  class IntegrityChecker::UnpublishingCheck
+    attr_reader :edition, :unpublishing
+
+    def initialize(edition, unpublishing)
+      @edition = edition
+      @unpublishing = unpublishing
+    end
+
+    def expected_type?
+      unpublishing["type"] == expected_type
+    end
+
+    def expected_type
+      return "withdrawal" if edition.withdrawn?
+      return "invalid edition state" unless edition.removed?
```

Comment:
Should this be a raise rather than return? I assume we don't expect this class to be called with a non withdrawn or removed edition?

Reply:
I don't expect it to be called with an edition that is not unpublished, possibly overly defensive code.  I can just delete it.

Reply:
Deleted